cpp/src/parquet/encryption/encryption_internal.cc

```diff
@@ -597,16 +617,27 @@ int AesDecryptor::AesDecryptorImpl::GcmDecrypt(span<const uint8_t> ciphertext,
   }
 
   // Setting additional authenticated data
+  if (aad.size() > static_cast<size_t>(std::numeric_limits<int>::max())) {
+    std::stringstream ss;
+    ss << "AAD size " << aad.size() << " overflows int";
+    throw ParquetException(ss.str());
+  }
   if ((!aad.empty()) && (1 != EVP_DecryptUpdate(ctx_, nullptr, &len, aad.data(),
                                                 static_cast<int>(aad.size())))) {
     throw ParquetException("Couldn't set AAD");
   }
 
   // Decryption
-  if (!EVP_DecryptUpdate(
-          ctx_, plaintext.data(), &len,
-          ciphertext.data() + length_buffer_length_ + kNonceLength,
-          ciphertext_len - length_buffer_length_ - kNonceLength - kGcmTagLength)) {
+  int32_t decryption_length =
+      ciphertext_len - length_buffer_length_ - kNonceLength - kGcmTagLength;
+  if (decryption_length > std::numeric_limits<int>::max()) {
```

Comment:
This doesn't look possible (how can a `int32_t` be larger than an `int`'s max value?)

Reply:
I mean there might have been platforms in the past where `int` is 16 bits, but those don't exist anymore, or are not supported by us in any case.

Reply:
Yeah the intention was to handle platforms with 16 bit ints, but you're right, this isn't needed. This should get compiled away when optimizations are enabled at least, but I'll remove it.